src/aws_cloud_hsm.erl

```diff
@@ -49,10 +49,10 @@
 %% partition group is a group of partitions that spans multiple physical
 %% HSMs.
 create_hapg(Client, Input)
-  when is_map(Client), is_map(Input) ->
+  when is_reference(Client), is_map(Input) ->
```

Comment:
This code is defined in the code generator:

  https://github.com/jkakar/aws-codegen/blob/master/priv/json.erl.eex#L17
  https://github.com/jkakar/aws-codegen/blob/master/priv/json.erl.eex#L20

We need to change it there.